src/Utilities/FileSystem.cpp

```diff
@@ -78,41 +78,40 @@ std::string get_absolute_path(const std::string& rel_path) {
           << rel_path << "'.");
       // LCOV_EXCL_STOP
     }
-    ERROR("Failed to get absolute path for an unknown reason: " << errno);
+    const auto local_errno = errno;
+    ERROR("Failed to get absolute path for an unknown reason: " << local_errno);
   }
   std::string return_name(name);
   free(name);
   return return_name;
 }
 
-void create_directory(const std::string& in_dir, const double wait_time,
+void create_directory(std::string dir, const double wait_time,
                       const size_t num_tries) {
   // each time we fail to create a directory, we increase the wait_time by this
   // factor
   static constexpr double wait_time_increase_factor = 1.1;
-  if (in_dir.empty()) {
+  if (dir.empty()) {
     ERROR("Cannot create a directory that has no name");
   }
-
-  std::string dir = filesystem::dirname(in_dir);
-  if ("/"s == dir) {
-    return;
+  if (std::string::npos == dir.find_first_not_of('/')) {
+    return;  // trying to create directory '/'
   }
-  dir += "/"s + filesystem::get_file_name(in_dir);
+  dir = filesystem::dirname(dir) + "/"s + filesystem::get_file_name(dir);
```

Comment:
It's not. For some reason I thought it was because SpEC hand the while loop removing the slashes but that loop was only needed because the old code did all the parsing inline. Removed now and added a few more slashes randomly in the test path to make sure that extra slashes do not interfere with correctness.